src/protocol/http/org/apache/jmeter/protocol/http/curl/BasicCurlParser.java

```diff
@@ -252,4 +821,232 @@ public Request parse(String commandLine) {
         }
         return result.toArray(new String[result.size()]);
     }
+    /**
+    *
+    * Set the username , password and baseurl of authorization
+    *
+    * @param authorizationStr the username and password of authorization
+    * @param url              the baseurl of authorization
+    * @param authorization    the object of authorization
+    */
+   public void setAuthUserPasswd(String authorizationStr, String url, Authorization authorization) {
+       String[] authorizationParameters = authorizationStr.split(":");
+       authorization.setUser(authorizationParameters[0].trim());
+       authorization.setPass(authorizationParameters[1].trim());
+       authorization.setURL(url);
+   }
+
+   /**
+    *
+    * Set the mechanism of authorization
+    *
+    * @param mechanism     the mechanism of authorization
+    * @param authorization the object of authorization
+    */
+   private void setAuthMechanism(String mechanism, Authorization authorization) {
+       switch (mechanism) {
+       case "basic":
+           authorization.setMechanism(Mechanism.BASIC);
+           break;
+       case "digest":
+           authorization.setMechanism(Mechanism.DIGEST);
+           break;
+       default:
+           break;
+       }
+   }
+
+   /**
+    *
+    * Set the parameters of proxy server in http request advanced
+    *
+    * @param request         http request
+    * @param proxyServerParameters the parameters of proxy server
+    *
+    */
+   private void setProxyServer(Request request, String proxyServerParameters) {
+       if (!proxyServerParameters.contains("://")) {
+           proxyServerParameters = "http://" + proxyServerParameters;
+       }
+       URI uriProxy = null;
+       try {
+           uriProxy = new URI(proxyServerParameters);
+           request.setProxyServer("scheme", uriProxy.getScheme());
+           Optional<String> userInfoOptional = Optional.ofNullable(uriProxy.getUserInfo());
+           if (userInfoOptional.isPresent()) {
+               String userinfo = userInfoOptional.get();
+               if (userinfo.contains(":")) {
+                   String[] userInfo = userinfo.split(":");
```

Comment:
What happens when the password has a `:` inside? Wouldn't we want to limit spliiting by using `userinfo.split(":", 1)`?
And why not call the variable `userInfoString`, to distinguish it from the later user `userInfo` String array?